api/core/routes.py

```diff
@@ -174,7 +163,143 @@ def get_ephemeris_by_name_jdstep():
+    lat = float(latitude)
+    lon = float(longitude)
+    ele = float(elevation)
+    
+    jd0 = float(startjd)
+    jd1 = float(stopjd) 
+    jds = float(stepjd)
+  
+    jd = my_arange(jd0,jd1,jds)
+
+    if(len(jd)>1000):
+        abort(400)
+
+    return create_result_list(lat, lon, ele, jd, name, False)
+
+@app.route('/ephemeris/catalog-number/')
+@limiter.limit("100 per second, 2000 per minute", key_func=lambda:get_forwarded_address(request))
+def get_ephemeris_by_catalog_number():
+    '''
+    Returns the Right Ascension and Declination relative to the observer's coordinates
+    for the given satellite's catalog number using the Two Line Element Data Set at inputted Julian Date.
+
+    **Please note, for the most accurate results, an inputted Julian Date close to the TLE epoch is necessary.
+
+    Parameters
+    ---------
+    catalog_number: 'str'
+        Satellite Catalog Number of object
+    latitude: 'float'
+        The observers latitude coordinate (positive value represents north, negative value represents south)
+    longitude: 'float'
+        The observers longitude coordinate (positive value represents east, negatie value represents west)
+    elevation: 'float'
+        Elevation in meters
+    julian_date: 'float'
+        UT1 Universal Time Julian Date. An input of 0 will use the TLE epoch.
+
+    Returns
+    -------
+    Name: 'str'
+        The name of the query object
+    JulianDate: 'float' or list of 'float'
+        UT1 Universal Time Julian Date. 
+    Right Ascension: 'float'
+        The right ascension of the satellite relative to observer coordinates in ICRS reference frame in degrees. Range of response is [0,360)
+    Declination: 'float'
+        The declination of the satellite relative to observer coordinates in ICRS reference frame in degrees. Range of response is [-90,90]
+    Altitude: 'float'
+        The altitude of the satellite relative to observer coordinates in ICRS reference frame in degrees. Range of response is [0,90]
+    Azimuth: 'float'
+        The azimuth of the satellite relative to observer coordinates in ICRS reference frame in degrees. Range of response is [0,360)
+    Range: 'float'
+        Range to object in km
+    '''
+
+    
+    catalog = request.args.get('catalog')
+    latitude = request.args.get('latitude')
+    longitude = request.args.get('longitude')
+    elevation = request.args.get('elevation')
+    julian_date = request.args.get('julian_date')
+
+    #check for mandatory parameters
+    if [x for x in (catalog, latitude, longitude, elevation, julian_date) if x is None]:
+        abort(400) 
+
+    #Cast the latitude, longitude, and jd to floats (request parses as a string)
+    lat = float(latitude)
+    lon = float(longitude)
+    ele = float(elevation)
+    
+    # Converting string to list
+    jul = str(julian_date).replace("%20", ' ').strip('][').split(', ')
+   
+    # Converting list elements to float
+    jd = [float(i) for i in jul]
```

Comment:
Please see if there is a utility in astropy.time that can do some or all of this so we are less likely to have some string parsing typo ruin our day

Reply:
So this is an html issue that Anthony was fighting with, where if you just enter something in the browser spaces can be turned into unsafe ASCII characters.